### ISSUE TITLE
Design and command tests failing when run in Visual Studio

### ISSUE THREAD

**Issue by ajcvickers** [dotnet/efcore#2082]
Normal development environment, using the .csproj files, TestDriven.NET as the runner.

```
------ Test started: Assembly: EntityFramework.Commands.FunctionalTests.dll ------

Test 'Microsoft.Data.Entity.Commands.ExecutorTest+SimpleProjectTest.GetMigrations_works_cross_domain' failed: System.AggregateException : One or more errors occurred.
---- System.IO.FileNotFoundException : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
---- The following constructor parameters did not have matching fixture data: SimpleProject project

    ----- Inner Stack Trace #1 (System.IO.FileNotFoundException) -----
    at Microsoft.Data.Entity.Relational.Design.Templating.Compilation.MetadataReferencesProvider..ctor(IServiceProvider serviceProvider)
    at Microsoft.Data.Entity.Commands.DatabaseTool..ctor(IServiceProvider serviceProvider, ILoggerProvider loggerProvider)
    at Microsoft.Data.Entity.Commands.Executor..ctor(Object logHandler, IDictionary args)
    ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----

Test 'Microsoft.Data.Entity.Commands.ExecutorTest+SimpleProjectTest.GetContextType_works_cross_domain' failed: System.AggregateException : One or more errors occurred.
---- System.IO.FileNotFoundException : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
---- The following constructor parameters did not have matching fixture data: SimpleProject project

    ----- Inner Stack Trace #1 (System.IO.FileNotFoundException) -----
    at Microsoft.Data.Entity.Relational.Design.Templating.Compilation.MetadataReferencesProvider..ctor(IServiceProvider serviceProvider)
    at Microsoft.Data.Entity.Commands.DatabaseTool..ctor(IServiceProvider serviceProvider, ILoggerProvider loggerProvider)
    at Microsoft.Data.Entity.Commands.Executor..ctor(Object logHandler, IDictionary args)
    ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----

Test 'Microsoft.Data.Entity.Commands.ExecutorTest+SimpleProjectTest.GetContextTypes_works_cross_domain' failed: System.AggregateException : One or more errors occurred.
---- System.IO.FileNotFoundException : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
---- The following constructor parameters did not have matching fixture data: SimpleProject project

    ----- Inner Stack Trace #1 (System.IO.FileNotFoundException) -----
    at Microsoft.Data.Entity.Relational.Design.Templating.Compilation.MetadataReferencesProvider..ctor(IServiceProvider serviceProvider)
    at Microsoft.Data.Entity.Commands.DatabaseTool..ctor(IServiceProvider serviceProvider, ILoggerProvider loggerProvider)
    at Microsoft.Data.Entity.Commands.Executor..ctor(Object logHandler, IDictionary args)
    ----- Inner Stack Trace #2 (Xunit.Sdk.TestClassException) -----

Test 'Microsoft.Data.Entity.Commands.ExecutorTest.GetContextType_works_with_multiple_assemblies' failed: System.IO.FileNotFoundException : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
    at Microsoft.Data.Entity.Relational.Design.Templating.Compilation.MetadataReferencesProvider..ctor(IServiceProvider serviceProvider)
    at Microsoft.Data.Entity.Commands.DatabaseTool..ctor(IServiceProvider serviceProvider, ILoggerProvider loggerProvider)
    at Microsoft.Data.Entity.Commands.Executor..ctor(Object logHandler, IDictionary args)

Test 'Microsoft.Data.Entity.Commands.ExecutorTest.GetMigrations_filters_by_context_name' failed: System.IO.FileNotFoundException : Could not load file or assembly 'Microsoft.CodeAnalysis, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
    at Microsoft.Data.Entity.Relational.Design.Templating.Compilation.MetadataReferencesProvider..ctor(IServiceProvider serviceProvider)
    at Microsoft.Data.Entity.Commands.DatabaseTool..ctor(IServiceProvider serviceProvider, ILoggerProvider loggerProvider)
    at Microsoft.Data.Entity.Commands.Executor..ctor(Object logHandler, IDictionary args)

62 passed, 5 failed, 0 skipped, took 4.27 seconds (xUnit.net 2.0.0 build 2880).
```

```
------ Test started: Assembly: EntityFramework.Relational.Design.Tests.dll ------

Test 'Microsoft.Data.Entity.Relational.Design.Tests.ApiConsistencyTest.Public_api_arguments_should_have_not_null_annotation' failed:
    System.IO.FileNotFoundException : Could not load file or assembly 'System.Reflection.Metadata, Version=1.0.19.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
    at System.Signature.GetSignature(Void* pCorSig, Int32 cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)
    at System.Reflection.RuntimeMethodInfo.FetchNonReturnParameters()
    at System.Reflection.RuntimeMethodInfo.GetParameters()
    ApiConsistencyTestBase.cs(59,0): at Microsoft.Data.Entity.ApiConsistencyTestBase.<>c__DisplayClass15.<Public_api_arguments_should_have_not_null_annotation>b__34(<>f__AnonymousType3`2 <>h__TransparentIdentifier2)
    at System.Linq.Enumerable.<SelectManyIterator>d__11`3.MoveNext()
    at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
    at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
    at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
    ApiConsistencyTestBase.cs(49,0): at Microsoft.Data.Entity.ApiConsistencyTestBase.Public_api_arguments_should_have_not_null_annotation()

Test 'Microsoft.Data.Entity.Relational.Design.Tests.ApiConsistencyTest.Async_methods_should_have_overload_with_cancellation_token_and_end_with_async_suffix' failed:
    System.IO.FileNotFoundException : Could not load file or assembly 'System.Reflection.Metadata, Version=1.0.19.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
    at System.Signature.GetSignature(Void* pCorSig, Int32 cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)
    at System.Reflection.RuntimeMethodInfo.get_ReturnType()
    ApiConsistencyTestBase.cs(81,0): at Microsoft.Data.Entity.ApiConsistencyTestBase.<>c__DisplayClass46.<Async_methods_should_have_overload_with_cancellation_token_and_end_with_async_suffix>b__55(<>f__AnonymousType0`2 <>h__TransparentIdentifier0)
    at System.Linq.Enumerable.<>c__DisplayClass2`1.<CombinePredicates>b__4(TSource x)
    at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
    at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
    at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
    ApiConsistencyTestBase.cs(76,0): at Microsoft.Data.Entity.ApiConsistencyTestBase.Async_methods_should_have_overload_with_cancellation_token_and_end_with_async_suffix()

1 passed, 2 failed, 0 skipped, took 0.97 seconds (xUnit.net 2.0.0 build 2880).
```

**Comment by lajones** [comment]
I’ll look into it. 
1.	What version of VS are you using?
2.	What version of TestDriven.NET?
3.	Does dnx . test at command line work for you?

**Comment by lajones** [comment]
Fixed by #2084.
